test/codeu/chat/server/BasicControllerTest.java

```diff
@@ -18,75 +18,109 @@
-  }
-
-  @Test
-  public void testAddUser() {
-
-    final User user = controller.newUser("user");
-
-    assertFalse(
-        "Check that user has a valid reference",
-        user == null);
-  }
-
-  @Test
-  public void testAddConversation() {
-
-    final User user = controller.newUser("user");
-
-    assertFalse(
-        "Check that user has a valid reference",
-        user == null);
-
-    final ConversationHeader conversation = controller.newConversation(
-        "conversation",
-        user.id);
-
-    assertFalse(
-        "Check that conversation has a valid reference",
-        conversation == null);
-  }
-
-  @Test
-  public void testAddMessage() {
-
-    final User user = controller.newUser("user");
-
-    assertFalse(
-        "Check that user has a valid reference",
-        user == null);
-
-    final ConversationHeader conversation = controller.newConversation(
-        "conversation",
-        user.id);
-
-    assertFalse(
-        "Check that conversation has a valid reference",
-        conversation == null);
-
-    final Message message = controller.newMessage(
-        user.id,
-        conversation.id,
-        "Hello World");
-
-    assertFalse(
-        "Check that the message has a valid reference",
-        message == null);
-  }
+    private Model model;
+    private BasicController controller;
+
+    @Before
+    public void doBefore() {
+        model = new Model();
+        controller = new Controller(Uuid.NULL, model);
+    }
+
+    @Test
+    public void testAddUser() {
+
+        final User user = controller.newUser("user");
+
+        assertFalse("Check that user has a valid reference", user == null);
+    }
+
+    @Test
+    public void testAddConversation() {
+
+        final User user = controller.newUser("user");
+
+        assertFalse("Check that user has a valid reference", user == null);
+
+        final ConversationHeader conversation = controller.newConversation("conversation", user.id);
+
+        assertFalse("Check that conversation has a valid reference", conversation == null);
+    }
+
+    @Test
+    public void testAddMessage() {
+
+        final User user = controller.newUser("user");
+
+        assertFalse("Check that user has a valid reference", user == null);
+
+        final ConversationHeader conversation = controller.newConversation("conversation", user.id);
+
+        assertFalse("Check that conversation has a valid reference", conversation == null);
+
+        final Message message = controller.newMessage(user.id, conversation.id, "Hello World");
+
+        assertFalse("Check that the message has a valid reference", message == null);
+    }
+
+    /**
+     * adds test for version function
+     */
+    @Test
+    public void testGetVersion() {
+       
+        String str = model.Version();
+        assertFalse("ERROR: Exception during call on server. Check log for details.", model == null);
+        
+        assertEquals("1.1", str);
```

Comment:
This is too specific to the current version. I would just do a not null or empty type check.